### PR TITLE
Spark supports k8s operator submit task

### PR DESCRIPTION
**Pull request by ChengJie1053**
Spark supports k8s operator submit task

Submit parameters as follows

{
    "executionContent":{
        "runType":"jar",
        "code":"show databases"
    },
    "params":{
        "configuration":{
            "startup":{
                "spark.executor.memory":"1g",
                "spark.driver.memory":"1g",
                "spark.executor.cores":"1",
                 "spark.app.main.class": "org.apache.spark.examples.SparkPi",
                "spark.app.name":"spark-submit-jar-cjtest123456",
                "spark.app.resource":"local:///opt/spark/examples/jars/spark-examples_2.12-3.2.1.jar",
                "spark.executor.instances":1,
                "spark.master":"k8s-operator",
                "linkis.spark.k8s.config.file":"/Users/chengjie/linkis/k8s_config/config",
                "linkis.spark.k8s.serviceAccount":"spark",
                "linkis.spark.k8s.master.url":"http://ip:port",
                "linkis.spark.k8s.username":"username",
                "linkis.spark.k8s.password":"password"
            }
        }
    },
    "source":{
        "scriptPath":"file:///tmp/hadoop/test.sql"
    },
    "labels":{
        "engineType":"spark-3.2.1",
        "engineConnMode":"once",
        "userCreator":"chengjie-IDE"
    }
}

There are two types of k8s connection, "spark.k8s.config.file" and "spark.k8s.master.url"

### CODE REVIEW

**Comment by peacewong** [comment]
ping @chenghuichen

**Comment by chenghuichen** [comment]
After all, `org.apache.linkis.engineplugin.spark.utils.SparkJobProgressUtil#getSparkJobInfo` should also be adapted for this submitting way.

**Comment by ChengJie1053** [comment]
> Besides, it would be nice to adapt `org.apache.linkis.engineplugin.spark.utils.SparkJobProgressUtil#getSparkJobInfo` to this submitting way.

Ok, thanks for reviewing the code

File: linkis-engineconn-plugins/spark/src/main/resources/spark-k8s-operator.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+kubectl delete serviceaccounts my-release-spark-operator --namespace spark-operator
+
+kubectl delete clusterrole my-release-spark-operator --namespace spark-operator
+
+kubectl delete clusterrolebindings my-release-spark-operator --namespace spark-operator
+```
+
+### 6. Submitting tasks with Restful API
+```text
+POST /api/rest_j/v1/entrance/submit
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/src/main/resources/spark-k8s-operator.md:45]: need to add linkis-cli demo

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/config/SparkConfiguration.scala [hunk 2]
```diff
@@ -50,6 +50,18 @@ object SparkConfiguration extends Logging {
   val SPARK_APP_RESOURCE = CommonVars[String]("spark.app.resource", "")
   val SPARK_APP_CONF = CommonVars[String]("spark.extconf", "")
 
+  val SPARK_K8S_CONFIG_FILE = CommonVars[String]("spark.k8s.config.file", "")
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/config/SparkConfiguration.scala:53]: If it is not a native parameter of spark or k8s, it need to add links prefix

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/factory/SparkEngineConnFactory.scala [hunk 3]
```diff
@@ -95,6 +95,17 @@ class SparkEngineConnFactory extends MultiExecutorEngineConnFactory with Logging
     sparkConfig.setJavaHome(variable(Environment.JAVA_HOME))
     sparkConfig.setSparkHome(SPARK_HOME.getValue(options))
     sparkConfig.setMaster(SPARK_MASTER.getValue(options))
+    sparkConfig.setK8sConfigFile(SPARK_K8S_CONFIG_FILE.getValue(options))
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/factory/SparkEngineConnFactory.scala:98]: Is there a need to judge Yarn or k8s here?

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/config/SparkConfiguration.scala (hunk 2 above)
Comment (ChengJie1053) [reply linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/config/SparkConfiguration.scala:53]: Ok, thanks for reviewing the code

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**